### ISSUE TITLE
Display names and text aren't separated by some space in TTY emacs

### ISSUE THREAD

**Issue by tsdh** [alphapapa/ement.el#82]
Here's a screenshot showing what I mean:
![Screenshot-2022-07-13_213111](https://user-images.githubusercontent.com/103854/178822869-9901b52d-b54e-4cb5-81e3-10dbf444f984.png)

The docstring of `ement-room-left-margin-width` says it should be set larger on TTYs than on gui emacsen, otherwise display names could run into the event text.  However, when I increase it, the margin becomes larger yet the names stay right-aligned and leave no space between name and message.

This becomes worse when names are truncated with ellipsis where it can happen that the ellipsis actually runs into the text:
![Screenshot-2022-07-13_220038](https://user-images.githubusercontent.com/103854/178824213-6ced3f8e-4b8e-42ff-8928-7abec5edccd3.png)

**Comment by alphapapa** [comment]
Thanks.  The issue is that, on GUI frames, the fringe is what separates the margin from the window body, but TTYs have no fringe.  When I last tested, it worked as advertised, but maybe something changed since then.

Probably the best solution now would be to use this recently added format in `ement-room-message-format-spec`:
```
(const :tag "IRC-style without margins, with wrap-prefix" "[%t] %S> %W%B%r")
```
But if you really do prefer the margin style, so that all names are aligned to the same column, then this bug will need to be fixed.

**Comment by tsdh** [comment]
Yeah, that style is better suited for a TTY frame, thanks.  And I don't think this is a bug but just a matter of fact that there are no fringes in a TTY.

One question which is probably related to the `%W` specifier: I'd like to be replied-to-text being a bit more distinguishable from the reply text.  See this screenshot:
![Screenshot-2022-07-15_070307](https://user-images.githubusercontent.com/103854/179154293-4badf6ff-81d2-4362-ad53-8f7ed5ddea9e.png)

It seems that replied-to-text always has a prefix whereas the reply-text has the same prefix for the first line but not for more lines.  I'd rather have replied-to-text prefixed (as it is now) and reply-to not prefixed so that the indentation clearly shows what's "the question" and what's the reply.

**Comment by alphapapa** [comment]
> Yeah, that style is better suited for a TTY frame, thanks. And I don't think this is a bug but just a matter of fact that there are no fringes in a TTY.

Well, it is a bug in any case, because the lack of fringes should be accounted for.  I just hadn't noticed it since I hadn't tested on a TTY recently.

> One question which is probably related to the `%W` specifier: I'd like to be replied-to-text being a bit more distinguishable from the reply text. See this screenshot: ![Screenshot-2022-07-15_070307](https://user-images.githubusercontent.com/103854/179154293-4badf6ff-81d2-4362-ad53-8f7ed5ddea9e.png)
> 
> It seems that replied-to-text always has a prefix whereas the reply-text has the same prefix for the first line but not for more lines. I'd rather have replied-to-text prefixed (as it is now) and reply-to not prefixed so that the indentation clearly shows what's "the question" and what's the reply.

I don't know why the non-reply part of the body isn't prefixed all the way to the end in that screenshot.  I haven't seen that happen in my testing.  Must be another bug.  Can you reproduce that on Emacs 28.1?

**Comment by tsdh** [comment]
Here's a screenshot of emacs 28 (basically -Q with just what's needed to get ement work):
![Screenshot-2022-07-16_172540](https://user-images.githubusercontent.com/103854/179361301-8fded8f3-230e-48e6-a094-74e58b976b07.png)
So here the prefixing of second-to-last line of the reply-text works.  It also works the same in a TTY frame.

Then I've tried again with my master emacs: it also works fine with -Q.  So long story short, it was caused by my config enabling `adaptive-wrap-prefix-mode` in `visual-line-mode`.

Anyway, now that the prefix is applied uniformly that makes it even harder to distinguish cited text from the reply, especially if the reply contains more than one level of nesting like it's the case in the last message of the screenshot where it's a conversation me, you, me again.  Maybe it would be a good approach to add some markers to the beginning of cited text, e.g., `>` like in email or `|`?

**Comment by alphapapa** [comment]
Ah, glad that it isn't a bug here, then.  :)

> Anyway, now that the prefix is applied uniformly that makes it even harder to distinguish cited text from the reply, especially if the reply contains more than one level of nesting like it's the case in the last message of the screenshot where it's a conversation me, you, me again.

Yes, AFAIK that's because the `shr` renderer also uses the wrap-prefix for the HTML blockquote that encloses the replied-to message, and the Ement formatter overrides the wrap prefix for the whole thing.

In order to fix that, I think we'd have to iterate across the text and add to any existing wrap-prefix properties rather than overwriting them.

> Maybe it would be a good approach to add some markers to the beginning of cited text, e.g., > like in email or |?

That could be a workaround, but I'd rather fix it "correctly" if possible, and as long as it doesn't cause a performance problem (e.g. when inserting many events at once).

**Comment by tsdh** [comment]
Obviously I'm all for the correct fix. ;-)

**Comment by vizs** [comment]
>Thanks. The issue is that, on GUI frames, the fringe is what separates the margin from the window body, but TTYs have no fringe. 

BTW, the bit about GUI frames is not always true.  You can set fringes-outside-margins to non-nil and it will no longer be after the margin.

**Comment by alphapapa** [comment]
@tsdh To solve this issue, I'd suggest that we set the format to the non-margin-using style by default when on a non-graphical display.  What do you think?

**Comment by tsdh** [comment]
@alphapapa I'd rather do nothing in that respect because it's quite common to have a single emacs process with both graphical and TTY frames.  It would be strange to change the layout based on in which frame was current when ement was loaded/started.

**Comment by alphapapa** [comment]
See also #9, which this may be a duplicate of.
